### PR TITLE
Commhistory implementation.

### PR DESCRIPTION
**Pull request by tswindell**

### CODE REVIEW

File: plugins/commhistory/src/calleventhandler.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    if(d->created) {
+        d->parent->eventModel()->modifyEvent(d->event);
+    } else {
+        d->created = d->parent->eventModel()->addEvent(d->event);
+    }
+
+    if(!d->created) WARNING_T("Failed to push event!");
+}
+
+void CallEventHandler::onEventsCommitted(QList<CommHistory::Event> events, bool success)
```
Comment (matthewvogt) [inline plugins/commhistory/src/calleventhandler.cpp]: "const QList &events"?
And then Q_UNUSED(events).

File: plugins/commhistory/src/calleventhandler.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+{
+    TRACE
+    Q_D(CallEventHandler);
+    if(success) d->committed = true;
```
Comment (matthewvogt) [inline plugins/commhistory/src/calleventhandler.cpp]: Committed doesn't seem to be used for anything?

**Comment by matthewvogt** [comment]
At least one of the commits needs a bug reference.

**Comment by matthewvogt** [comment]
LGTM

**Comment by jpetrell** [comment]
Sorry totally missed these pull requests getting merged. I work in Phone area, the tasks were marked under VOIP.

This is late, but some comments anyway:
- if call logging is moved to voicecall why is StreamChannelListener still run on CommHistory side?
- outbound calls also get marked as missed calls, which is behavior a break
- missed call counter is broken, after each missed call the eventCount value increases by two, I wonder if it is because of two parties logging calls (StreamChannelListener, CallEventHandler)
- did testing team do sanity testing / execute call tests?

**Comment by jpetrell** [comment]
Reverted for now. Created pull request to re-enable https://git.merproject.org/mer-core/voicecall/merge_requests/5 , also fixed the "outbound calls also get marked as missed calls" issue.

**Pull request merged**